Python 3: print statement -> print function

josephsl opened the pull request:
### Link to issue number:
None

### Summary of the issue:
In python 3, print keyword is now a function.

### Description of how this pull request fixes the issue:
Changes occurrence of "print" statement to a function. Note that this one is also applicable to Python 2, but it is based on Project Threshold/Python 3 assumptions.

### Testing performed:
Tested multiple times on source copy of Python 3 NVDA, Python 2 and 3 interpreter, and current NVDA alpha release.

### Known issues with pull request:
None

### Change log entry:
None

josephsl committed:
App module handler/Python 3: print statement -> function when printing crash dump path.

josephsl committed:
SAPI5 synth driver/Python 3: print statement -> function when announcing function hooks.

michaelDCurran commented on source/synthDrivers/sapi5.py:
```diff
@@ -37,9 +37,9 @@ class FunctionHooker(object):
 		hook=NVDAHelper.localLib.dllImportTableHooks_hookSingle(targetDll,importDll,funcName,newFunction)
 		if hook:
-			print "hooked %s"%funcName
+			print("hooked %s"%funcName)
```
Actually, the print statements in this file should have always been log.debug calls. Currently they are just noise going to stdout.

josephsl commented:
Hi, I was wondering for a while why there was a print statement… now we know – change will be done in a moment. Thanks for this clarification.

 

From: Michael Curran <notifications@github.com> 
Sent: Tuesday, June 4, 2019 4:17 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Python 3: print statement -> print function (#9664)

 

@michaelDCurran requested changes on this pull request.

  _____  

In source/synthDrivers/sapi5.py <https://github.com/nvaccess/nvda/pull/9664#discussion_r290532101> :

> @@ -37,9 +37,9 @@ class FunctionHooker(object):
        def __init__(self,targetDll,importDll,funcName,newFunction):
         hook=NVDAHelper.localLib.dllImportTableHooks_hookSingle(targetDll,importDll,funcName,newFunction)
               if hook:
-                      print "hooked %s"%funcName
+                      print("hooked %s"%funcName)

Actually, the print statements in this file should have always been log.debug calls. Currently they are just noise going to stdout.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/9664?email_source=notifications&email_token=AB4AXEC7T2LZOXKUFHVTX6LPY3ZXTA5CNFSM4HSDUU4KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB2S465I#pullrequestreview-245747573> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4AXEHWAF447XINVLQLGTLPY3ZXTANCNFSM4HSDUU4A> .

josephsl committed:
SAPI 5: print function -> log.debug.

Reviewed and clarified by Mick Curran (NV Access): print function should have been a log.debug call.

michaelDCurran reviewed (changes requested):
I would prefer this to go straight to threshold rather than master, unless @feerrenrut  instructed otherwise?
Reasoning is that although 
```
print("hello")
```
Does syntactitally work in Python 2, this is more just a side afect due to  the print statement allowing no whitespace between the keyword and a following tuple.
If you try to do things like provide multiple arguments, you see that it literally prints a tuple, and if you try giving it Python 3 keyword arguments, you get a syntax error.
So I think adding it to master now is miss-leading.
Of course in Python 2 we could do:
```
from __future__ import print_function
```
   But we'll then have to remove that line as soon as threshold is merged to master anyway.

josephsl commented:
Hi, CC @LeonardDer

 

From: Michael Curran <notifications@github.com> 
Sent: Tuesday, June 4, 2019 5:02 PM
To: nvaccess/nvda <nvda@noreply.github.com>
Cc: Joseph Lee <joseph.lee22590@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [nvaccess/nvda] Python 3: print statement -> print function (#9664)

 

@michaelDCurran requested changes on this pull request.

I would prefer this to go straight to threshold rather than master, unless @feerrenrut <https://github.com/feerrenrut>  instructed otherwise?
Reasoning is that although

print("hello")

Does syntactitally work in Python 2, this is more just a side afect due to the print statement allowing no whitespace between the keyword and a following tuple.
If you try to do things like provide multiple arguments, you see that it literally prints a tuple, and if you try giving it Python 3 keyword arguments, you get a syntax error.
So I think adding it to master now is miss-leading.
Of course in Python 2 we could do:

from __future__ import print_function

But we'll then have to remove that line as soon as threshold is merged to master anyway.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/nvaccess/nvda/pull/9664?email_source=notifications&email_token=AB4AXECLUI2P5MHE2XXOB5DPY367HA5CNFSM4HSDUU4KYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB2S7P3A#pullrequestreview-245757932> , or mute the thread <https://github.com/notifications/unsubscribe-auth/AB4AXECRQZQ7UWEG6PKSWX3PY367HANCNFSM4HSDUU4A> .

josephsl commented:
Hi,

Done - sorry for rebases and forced commits.

Thanks.

leonardder commented:
Mick has a valid point here. Switching back to Threshold makes sense.

feerrenrut reviewed (approved):
Looks good, also on this PR, please outline the process used.

josephsl commented:
Hi,

Process used:

1. Did grep on print statement (grep -lr "print " source).
2. For each occurrence, changed the statement to a function.
3. For SAPI5 module, changed from print statement to a log.debug function.

Thanks.
